### ISSUE TITLE
multiple-process duplicate results

### ISSUE THREAD

**Issue by BaiBenny** [optuna/optuna#5058]
### Don't use GitHub Issues to ask support questions.

Hello EveryOne:
    Recently, I am using Optuna for hyperparameter optimization tasks, but I need to use multiple processes (not multi-threads). I tried using the ProcessPoolExecutor implementation, but after running it I found there were duplicates in the results. code show as below:

``` python
from concurrent.futures import ProcessPoolExecutor
from optuna.trial import TrialState
import optuna
import os
import sklearn.datasets
import sklearn.linear_model
import sklearn.model_selection
from optuna.study import MaxTrialsCallback
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split

def objective(trial: optuna.trial.FrozenTrial, ):
    print("==========pid: ", os.getpid())
    regression_name = trial.suggest_categorical('regression', ['SVR', 'RandomForest'])
    if regression_name == 'SVR':
        svr_c = trial.suggest_float('svr_c', 1e-10, 1e10, log=True)
        regression_obj = sklearn.svm.SVR(C=svr_c)
    else:
        rf_max_depth = trial.suggest_int('rf_max_depth', 2, 32)
        regression_obj = RandomForestRegressor(max_depth=rf_max_depth, n_jobs=1)
    data = sklearn.datasets.load_diabetes(as_frame=True)
    x_train, x_val, y_train, y_val = train_test_split(data['data'], data['target'], random_state=0)
    regression_obj.fit(x_train, y_train)
    y_pred = regression_obj.predict(x_val)
    mse = sklearn.metrics.mean_squared_error(y_val, y_pred)
    return mse

def run(study):
    study.optimize(objective, callbacks=[MaxTrialsCallback(10, (TrialState.FAIL, TrialState.COMPLETE, TrialState.RUNNING, TrialState.PRUNED))], catch=ValueError)

if __name__ == '__main__':
    study_name = "example-study1"  # Unique identifier of the study.
    storage = optuna.storages.JournalStorage(
        optuna.storages.JournalFileStorage("/mnt/disk1/hubaiqing/tmp/journal.log"),
    )
    study = optuna.create_study(direction='minimize', storage=storage)
    n_jobs = 2
    executor = ProcessPoolExecutor(max_workers=n_jobs)
    for i in range(n_jobs):
        executor.submit(run, study)
    executor.shutdown()
```

The running results are as follows:
``` shell
[I 2023-10-20 14:29:40,171] A new study created in Journal with name: no-name-75ee2904-3ec3-4ee2-b9c5-dbb327c7fc54
[I 2023-10-20 14:29:40,815] Trial 0 finished with value: 11580.817623987665 and parameters: {'regression': 'SVR', 'svr_c': 350878.9915565393}. Best is trial 0 with value: 11580.817623987665.
[I 2023-10-20 14:29:40,949] Trial 1 finished with value: 11580.817623987665 and parameters: {'regression': 'SVR', 'svr_c': 350878.9915565393}. Best is trial 0 with value: 11580.817623987665.
[I 2023-10-20 14:29:41,083] Trial 2 finished with value: 6978.1764271809625 and parameters: {'regression': 'SVR', 'svr_c': 4531.925776977921}. Best is trial 2 with value: 6978.1764271809625.
[I 2023-10-20 14:29:41,217] Trial 3 finished with value: 6978.1764271809625 and parameters: {'regression': 'SVR', 'svr_c': 4531.925776977921}. Best is trial 2 with value: 6978.1764271809625.
[I 2023-10-20 14:29:41,485] Trial 4 finished with value: 11580.817623987665 and parameters: {'regression': 'SVR', 'svr_c': 61058.345299578716}. Best is trial 2 with value: 6978.1764271809625.
[I 2023-10-20 14:29:41,611] Trial 5 finished with value: 11580.817623987665 and parameters: {'regression': 'SVR', 'svr_c': 61058.345299578716}. Best is trial 2 with value: 6978.1764271809625.
[I 2023-10-20 14:29:41,787] Trial 7 finished with value: 5156.67306206683 and parameters: {'regression': 'SVR', 'svr_c': 9.988901594482554e-10}. Best is trial 7 with value: 5156.67306206683.
[I 2023-10-20 14:29:41,912] Trial 8 finished with value: 4554.276268314136 and parameters: {'regression': 'SVR', 'svr_c': 0.6196930326659138}. Best is trial 8 with value: 4554.276268314136.
[I 2023-10-20 14:29:42,147] Trial 6 finished with value: 5156.67306206683 and parameters: {'regression': 'SVR', 'svr_c': 9.988901594482554e-10}. Best is trial 8 with value: 4554.276268314136.
[I 2023-10-20 14:29:42,272] Trial 9 finished with value: 11580.817623987665 and parameters: {'regression': 'SVR', 'svr_c': 199885929.8485345}. Best is trial 8 with value: 4554.276268314136.
```

The above results seem to complete multiple processes, but the actual results are repeated. For example Trial 0 and Trial 1.

So what should I do to truly implement multi-process running of multiple trials?

**Comment by BaiBenny** [comment]
@nzw0301 Hello , why closed the issue? If you know the cause or solution to this problem, please reply, thank you very much.

**Comment by nzw0301** [comment]
The reasons were as follows:

1. As in [the issue description](https://github.com/optuna/optuna/issues/5058#issue-1953673982),

> Don't use GitHub Issues to ask support questions.

2. You have already asked the same question as https://github.com/optuna/optuna/discussions/5059, so I did not need to convert this issue as a discussion post.

Thus I've closed this issue.

I understand this separation might not be user friendly, but this is necessary for dev team's issue tracking to develop optuna smoothly. 

Best regards,
